### PR TITLE
[FLINK-33899][connectors/mongodb] Java 17 and 21 support for mongodb connector

### PR DESCRIPTION
**Pull request by Jiabao-Sun**
After [FLINK-33302](https://issues.apache.org/jira/browse/FLINK-33302) is finished it is now possible to specify jdk version
That allows to add jdk17 and jdk21 support.

### CODE REVIEW

File: flink-connector-mongodb/pom.xml [hunk 1]
```diff
@@ -32,6 +32,13 @@ under the License.
 
 	<packaging>jar</packaging>
 
+	<properties>
+		<surefire.module.config>
+			--add-opens=java.base/java.util=ALL-UNNAMED
+			--add-opens=java.base/java.lang=ALL-UNNAMED
+		</surefire.module.config>
```
Comment (leonardBang) [inline flink-connector-mongodb/pom.xml:39]: Could you explain a little why we need this ?

File: .github/workflows/weekly.yml [hunk 2]
```diff
@@ -34,25 +34,26 @@ jobs:
...
           flink: 1.17.1,
-          branch: v1.0
-        },{
-          flink: 1.18.0,
-          branch: v1.0
+          branch: v3.1
         }, {
-          flink: 1.19-SNAPSHOT,
-          branch: v1.0
+          flink: 1.18.0,
```
Comment (leonardBang) [inline .github/workflows/weekly.yml:50]: Could we bump the flink version to latest minor version?

File: .github/workflows/weekly.yml [hunk 3]
```diff
@@ -34,25 +34,26 @@ jobs:
...
+          jdk: '8, 11, 17',
           branch: main
         }, {
           flink: 1.19-SNAPSHOT,
+          jdk: '8, 11, 17, 21',
           branch: main
         }, {
           flink: 1.16.2,
-          branch: v1.0
+          branch: v3.1
```
Comment (leonardBang) [inline .github/workflows/weekly.yml:45]: Do we have a branch named `v3.1` ?

File: .github/workflows/push_pr.yml [hunk 4]
```diff
@@ -25,7 +25,14 @@ jobs:
...
     strategy:
       matrix:
-        flink: [1.16-SNAPSHOT, 1.17-SNAPSHOT, 1.18-SNAPSHOT, 1.19-SNAPSHOT]
+        flink: [ 1.16-SNAPSHOT, 1.17-SNAPSHOT ]
+        jdk: [ '8, 11' ]
+        include:
+          - flink: 1.18-SNAPSHOT
+            jdk: '8, 11, 17'
+          - flink: 1.19-SNAPSHOT
+            jdk: '8, 11, 17, 21'
```
Comment (leonardBang) [inline .github/workflows/push_pr.yml:34]: I'm not sure current matrix is you want or not, maybe you need something like if I understand azure pipeline yaml correctly?
```
        flink: [1.16-SNAPSHOT, 1.17-SNAPSHOT, 1.18-SNAPSHOT, 1.19-SNAPSHOT]
        jdk: ['8, 11, 17, 21']
        exclude:
          - flink: 1.16-SNAPSHOT
            jdk: '17, 21'
          - flink: 1.17-SNAPSHOT
            jdk: '21'
```

Comment (Jiabao-Sun) [reply .github/workflows/push_pr.yml:34]: This modification refers to the jdbc-connector.

https://github.com/apache/flink-connector-jdbc/blob/e3dd84160cd665ae17672da8b6e742e61a72a32d/.github/workflows/push_pr.yml#L19-L38

**Review by leonardBang — APPROVED** [review]
Thanks @Jiabao-Sun for addressing comments, LGTM

**Comment by leonardBang** [comment]
@Jiabao-Sun could you help rebase this PR to latest master?

**Comment by Jiabao-Sun** [comment]
Thanks @leonardBang for the review.
Rebased.

**Pull request merged**